### PR TITLE
feat(connector): [Fiserv] Currency Unit Conversion

### PR DESCRIPTION
**Pull request by SagarDevAchar**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Addressing Issue: #2225 
- Modified two files in `hyperswitch/crates/router/src/connector/`
  - `fiserv.rs`
    - Implement `get_currency_unit` function
    - Modify `ConnectorIntegration` implementations for `Fiserv`
  - `fiserv/transformers.rs`
    - Implement `FiservRouterData<T>` structure and functionality
    - Modify implementation for `FiservPaymentsRequest` and `FiservRefundRequest`

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

We need to create a payment using Fiserv and test the currency unit from connector response and connector dashboard.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by AkshayaFoiger — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/fiserv/transformers.rs [hunk 1]
```diff
@@ -389,35 +423,44 @@ pub struct SessionObject {
...
             .ok_or(errors::ConnectorError::RequestEncodingFailed)?;
         let session: SessionObject = metadata
             .parse_value("SessionObject")
             .change_context(errors::ConnectorError::RequestEncodingFailed)?;
-        let amount =
-            utils::to_currency_base_unit(item.request.amount_to_capture, item.request.currency)?;
+        let amount = utils::to_currency_base_unit(
+            item.router_data.request.amount_to_capture,
+            item.router_data.request.currency,
+        )?;
```
Comment (AkshayaFoiger) [inline crates/router/src/connector/fiserv/transformers.rs:443]:
```suggestion
```

File: crates/router/src/connector/fiserv/transformers.rs [hunk 2]
```diff
@@ -389,35 +423,44 @@ pub struct SessionObject {
...
         Ok(Self {
             amount: Amount {
                 total: amount,
```
Comment (AkshayaFoiger) [inline crates/router/src/connector/fiserv/transformers.rs:446]:
```suggestion
                total: item.amount.clone(),
```

**Review by AkshayaFoiger — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/fiserv/transformers.rs (hunk 2 above)
Comment (SagarDevAchar) [reply crates/router/src/connector/fiserv/transformers.rs]: Resolved! Thanks @AkshayaFoiger!

**Comment by AkshayaFoiger** [comment]
LGTM!

**Review by AkshayaFoiger — APPROVED** [review]

**Review by swangi-kumari — APPROVED** [review]
LGTM

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @SagarDevAchar!

**Comment by deepanshu-iiitu** [comment]
Hey @SagarDevAchar  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**